File: extract_holes.py
import re

with open('public/logo_previous.svg', 'r') as f:
    content = f.read()

path1_match = re.search(r'<path[^>]*d="(.*?)"', content)
if not path1_match:
    print("No path found")
    exit()

path1 = path1_match.group(1)
subpaths = [s.strip() for s in path1.replace('Z M', 'Z|M').split('|')]

print(f'Total subpaths in path 1: {len(subpaths)}')

out_html = "<html><body><svg viewBox='70 180 900 620' style='width:900px; height:620px; border:1px solid black;'>\n"
out_html += f"  <path fill='#3D3D3D' d='{subpaths[0]}' />\n"

import random

for i, sp in enumerate(subpaths[1:]):
    color = f"#{random.randint(0, 0xFFFFFF):06x}"
    out_html += f"  <path id='hole_{i+1}' fill='{color}' d='{sp}' onclick='alert(this.id)' />\n"

out_html += "</svg></body></html>"

with open('public/holes_viewer.html', 'w') as f:
    f.write(out_html)

print("Created public/holes_viewer.html")

File: extract_holes.js
const fs = require('fs');

const content = fs.readFileSync('public/logo.svg', 'utf8');

const path1Match = content.match(/<path[^>]*d="(.*?)"/);
if (!path1Match) {
    console.log("No path found");
    process.exit();
}

const path1 = path1Match[1];
const subpaths = path1.replace(/Z M/g, 'Z|M').split('|').map(s => s.trim());

console.log(`Total subpaths in path 1: ${subpaths.length}`);

let outHtml = "<html><body><svg viewBox='70 180 900 620' style='width:900px; height:620px; border:1px solid black;'>\n";
outHtml += `  <path fill='#3D3D3D' d='${subpaths[0]}' />\n`;

for (let i = 1; i < subpaths.length; i++) {
    const color = '#' + Math.floor(Math.random()*16777215).toString(16).padStart(6, '0');
    outHtml += `  <path id='hole_${i}' fill='${color}' d='${subpaths[i]}' onclick='alert(this.id)' />\n`;
}

outHtml += "</svg></body></html>";

fs.writeFileSync('public/holes_viewer.html', outHtml);
console.log("Created public/holes_viewer.html");
